File: Python/beginner/1165.py
# https://judge.beecrowd.com/pt/problems/view/1165
# Números Primos

# Asking for the number of test cases
from math import sqrt


testCases = int(input())

# Iterating through test cases
for _ in range(testCases):
    # Getting number
    number = int(input())
    
    # Below 2 are not primes
    if number < 2:
        print(f'{number} nao eh primo')
    
    # Testing other cases
    else:
        # Boolean Guard to check if number is prime
        is_Prime = True
        # Iterating between 2 and up to sqrt(number)        
        for i in range(2, round(sqrt(number)) + 1):
            if number % i == 0:
                is_Prime = False
                break
            
        # Printing   
        print(f'{number} eh primo' if is_Prime else f'{number} nao eh primo') 
    
# By Fernando Serra
# https://github.com/fernandosserra

File: C/beginner/1165.c
// https://judge.beecrowd.com/pt/problems/view/1165
// Números Primos

// Libraries
#include <stdio.h>
#include <math.h>

// Main
int main(){
    // Variables
    int testCases, usrInput;

    // Getting Test Cases
    scanf("%d", &testCases);

    // Iterating
    for (int i = 0; i < testCases; i++){
        scanf("%d", &usrInput);

        int isPrime = 1; // Assuming it is prime

        if (usrInput < 2) {
            isPrime = 0; // Below 2 are not primes
        } else {
            // To check if a number is prime, we only need to check its divisors up to its square root
            for (int j = 2; j <= sqrt(usrInput); j++){
                if (usrInput % j == 0){
                    isPrime = 0;
                    break;
                }
            }
        }

        // Printing our conclusions
        if (isPrime){
            printf("%d eh primo\n", usrInput);
        } else {
            printf("%d nao eh primo\n", usrInput);
        }
    }

    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra